myapp/app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,34 @@
+class SessionsController < ApplicationController
+  skip_before_action :require_log_in, only: [:new, :create]
+  before_action :verify_email, only: [:create]
+
+  def new; end
+
+  def create
+    if @user.authenticate(session_params[:password])
+      session[:user_id] = @user.id
+      redirect_to root_path
+    else
+      flash.now[:danger] = t('sessions.flash.login.invalid_password')
+      render :new
+    end
+  end
+
+  def destroy
+    session.delete(:user_id)
+    redirect_to login_path
+  end
+
+  private
+
+  def verify_email
+    @user = User.find_by!(email: session_params[:email])
+  rescue StandardError
+    flash.now[:danger] = t('sessions.flash.login.invalid_email')
+    render :new
```

Comment:
このくらいの処理であれば例外処理にしなくても良さそうです！
（StandardErrorでキャッチしてしまうとレコードが見つからなかった時以外もキャッチされてしまうため。）
```suggestion
    @user = User.find_by(email: session_params[:email])
    return if @user.present?

    flash.now[:danger] = t('sessions.flash.login.invalid_email')
    render :new
```

Reply:
上記ご指摘の通り、不必要な例外処理は避けるべきですね。。
以降の実装でも注意致します。

またverify_emailをprivateメソッドとして切り出しておりますが、以下の点を踏まえメソッドを呼び出し元に統一しました。
・現状として呼び出し元が1ヶ所(create)だけであること
・before_actionを使用せずにメソッド内から呼び出す場合、呼び出し元メソッド(create)を抜ける為に追加の判定処理が必要になること